lightningd/payalgo.c

```diff
@@ -435,7 +440,7 @@ static void json_pay_getroute_reply(struct subd *gossip UNUSED,
 	 * That loss will not be representable in double. So, it's Okay to
 	 * cast u64 to double for feepercent calculation. */
 	feepercent = ((double) fee) * 100.0 / ((double) pay->msatoshi);
-	fee_too_high = (feepercent > pay->maxfeepercent);
+	fee_too_high = (fee <= pay->exemptfee && feepercent > pay->maxfeepercent);
```

Comment:
I think it should be `fee_too_high = (fee > pay->exemptfee && feepercent > pay->maxfeepercent);`

`fee_too_high` should only be true if it is more than the `pay->exemptfee`

Reply:
Doh :+1: